### DISCUSSION TITLE
Refine the "plugins" concept to be distinct from tools

### DISCUSSION THREAD

**Discussion by syhol** [jdx/mise#3156] [Ideas]
Original title: Deprecate and remove "plugins" concept and fully move to aliases

It seem that mise is [leaning](https://github.com/jdx/mise/pull/2979) more [heavily](https://github.com/jdx/mise/pull/3067) on the concepts of "tools, aliases, and backends" vs "traditional asdf & vfox plugins", and I was wondering if you want to go all in to make mise simpler to understand and work with, or continue support the concept of plugins.

This includes:
- _Removing_ the terms "plugin" from the docs
- _Removing_ the plugin cli commands
- _Keeping or tweaking_ the internals and directories for plugins

Problems of having plugins concept in mise today:
- There are lots of concepts to learn: backend, plugin, tool, alias, registry.
- Plugins aren't orthogonal with other mise concepts: They're almost tools but only for some backends and `mise plugin add x y` creates something like an alias but not quite.
- Plugins fall out of date and newer tool releases can't be installed

Challenges of removing the plugins concept:
- Since plugins would still exist under the hood, how/when would they be updated?
- How would "generic asdf plugins" like https://github.com/amrox/asdf-pyapp work, but it [doesn't seem to work](https://discord.com/channels/1066429325269794907/1066429326687473776/1309627595964547072) very well today anyway

**Comment by jdx** [comment]
Plugins definitely aren't going away. You can have plugins that only modify env vars and don't provide tools.

In asdf, the concept of tools and plugins were one in the same. mise started with that but now that we have backends the concepts need to be separated.

I actually don't think this is going to be that confusing, except for some places in the docs/code where "plugins" is still conflated to mean "tool".

I've been able to move a lot of the tools over to aqua, especially more popular ones. The exposure to asdf plugins should diminish over time as we can leverage aqua/ubi more.

"Plugins" is going to be a term specific to extending functionality in mise, whether that's adding a tool or doing something else. I think this is more typical of the term generally anyhow.

**Reply by syhol** [reply]
Nice. thanks for the insight.

This problem has been greatly mitigated with the following:
- https://github.com/jdx/mise/commit/0556024b5abdb2d5f1cb025d105494c71aa79647
- https://github.com/jdx/mise/commit/1a38802dd2c729805654638a6e2464afed6e8e14
- https://github.com/jdx/mise/commit/8ca6f5f9e8df0be7b714ffe6d030fd60bf04fcd7
- https://github.com/jdx/mise/commit/c82d4d7e16cd79a6c6cab759065f0ec0d9d2badd

Happy to close this discussion 😄
